### PR TITLE
refactor: Simplify WAD server controller logic

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
The previous implementation had too many hacks. Rewrote it in a cleaner and more stable manner

### CODE REVIEW

File: lib/winappdriver.js [hunk 1]
```diff
@@ -1,150 +1,191 @@
...
-      this.emit(WinAppDriver.EVENT_ERROR, e);
-      // just because we had an error doesn't mean the WinAppDriver process
-      // finished; we should clean up if necessary
-      if (processIsAlive) {
-        await this.proc.stop();
-      }
-      log.errorAndThrow(e);
+  async stop () {
+    if (this.isRunning) {
+      await this.proc.stop('SIGTERM');
```
Comment (KazuCocoa) [inline lib/winappdriver.js:88]: It seems SIGINT, SIGKILL and SIGTERM have no differences on Windows...?
https://nodejs.org/docs/latest-v10.x/api/process.html

File: lib/winappdriver.js [hunk 2]
```diff
@@ -1,150 +1,191 @@
...
+}
 
-    return this.jwproxy.sessionId;
+const RUNNING_PROCESS_IDS = [];
+process.once('exit', () => {
+  for (const pid of RUNNING_PROCESS_IDS) {
+    const command = `taskkill.exe /PID ${pid}`;
+    try {
+      execSync(command);
+    } catch (ign) {}
```
Comment (KazuCocoa) [inline lib/winappdriver.js:107]: Probably it's better to print a log to report us when taskkill.exe does not work without `\F` (current case)

File: lib/winappdriver.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/winappdriver.js:88]: yep, makes sense

File: lib/winappdriver.js (hunk 2 above)
Comment (mykola-mokhnach) [reply lib/winappdriver.js:107]: added

**Review by KazuCocoa — APPROVED** [review]

**Pull request merged**